core/src/main/java/com/linecorp/armeria/server/logging/structured/StructuredLog.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured;
+
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.ResponseLog;
+import com.linecorp.armeria.common.util.DateTimeFormat;
+import com.linecorp.armeria.common.util.UnitFormatter;
+
+/**
+ * A representation and constructor of a service log which holds only very common fields that are protocol
+ * agnostic.
+ */
+public abstract class StructuredLog {
```

Comment:
I think it isn't necessary if the purpose to do that is : https://github.com/line/armeria/pull/324#discussion_r87324162